### PR TITLE
Idaho PBF Modification

### LINKED ISSUE #3179: Add Idaho Permanent Building Fund Tax
The absence of this small Idaho tax makes all the test cases have a tax difference in the TAXSIM35 integration testing described in issue #993.   What do people think about whether or not this tax should be considered part of the Idaho state income tax?

Screenshot 2023-10-20 at 4 01 29 PM

### PR DESCRIPTION
**Pull request by shuyu7zhan**
Fixes #3179

copilot:all

### CODE REVIEW

File: policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        # p = parameters(period).gov.states.id.tax.income.other_taxes.pbf
+
+        # eligible if income less than filing status specified amount
+        income_ineligible = (
+            tax_unit("id_income_tax_before_non_refundable_credits", period)
+            <= 0
+        )
+
+        # eligible if receiving public assistance tanf
+        tanf_received = tax_unit("tanf", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py:23]:
```suggestion
        tanf_received = tax_unit.spm_unit("tanf", period) > 0
```

File: policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        tanf_ineligible = tanf_received > 0
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py:24]:
```suggestion
```

File: policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+        # eligible if head or spouse is blind
+        blind_head = tax_unit("blind_head", period)
+        blind_spouse = tax_unit("blind_spouse", period)
+        blind_ineligible = blind_head | blind_spouse
+
+        return ~income_ineligible & ~tanf_ineligible & ~blind_ineligible
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py:31]:
```suggestion
        return ~income_ineligible & ~tanf_received & ~blind_ineligible
```

File: policyengine_us/tests/policy/baseline/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.yaml
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.yaml:1]: need to break out the test to add som_unit level inputs

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf.py:1]: add this value to the `id_income_tax_before_refundable_credits` calculation 

add it to the test

File: policyengine_us/parameters/gov/states/id/tax/income/other_taxes/pbf/age_threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/other_taxes/pbf/age_threshold.yaml:1]: delete this file

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/id/tax/income/other_taxes/pbf/amount.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+description: Idaho requires filers to pay a permanent building fund tax of this amount.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/other_taxes/pbf/amount.yaml:1]:
```suggestion
description: Idaho levies a permanent building fund tax of this amount.
```

File: policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+from policyengine_us.model_api import *
+
+
+class id_pbf_requirement(Variable):
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py:4]:
```suggestion
class id_pbf_liable(Variable):
```

File: policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    value_type = bool
+    entity = TaxUnit
+    label = "Required to pay the Idaho permanent building fund tax"
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py:7]:
```suggestion
    label = "Liable for the Idaho permanent building fund tax"
```

File: policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    definition_period = YEAR
+    reference = "https://tax.idaho.gov/wp-content/uploads/forms/EIN00046/EIN00046_11-15-2021.pdf#page=10"
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        # Not required to pay the PBF if there is no income tax
+        income_tax_payment = (
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py:14]:
```suggestion
        owes_income_tax_before_credits = (
```

File: policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+            tax_unit("id_income_tax_before_non_refundable_credits", period) > 0
+        )
+
+        # Not required to pay the PBF if receiving public assistance, tanf
+        tanf_received = tax_unit.spm_unit("tanf", period) > 0
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_requirement.py:19]:
```suggestion
        receives_tanf = tax_unit.spm_unit("tanf", period) > 0
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_liable.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    label = "Liable for the Idaho permanent building fund tax"
+    definition_period = YEAR
+    reference = "https://tax.idaho.gov/wp-content/uploads/forms/EIN00046/EIN00046_11-15-2021.pdf#page=10"
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        # Not required to pay if there is no income tax
+        owes_income_tax_before_credits = (
+            tax_unit("id_income_tax_before_non_refundable_credits", period)
+            <= 0
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_liable.py:16]: and flip line 28, or change the variable name. `owes` indicates that tax > 0
```suggestion
            > 0
```

File: policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_liable.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    entity = TaxUnit
+    label = "Liable for the Idaho permanent building fund tax"
+    definition_period = YEAR
+    reference = "https://tax.idaho.gov/wp-content/uploads/forms/EIN00046/EIN00046_11-15-2021.pdf#page=10"
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        # Not required to pay if there is no income tax
+        owes_income_tax_before_credits = (
+            tax_unit("id_income_tax_before_non_refundable_credits", period)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/id/tax/income/other_taxes/pbf/id_pbf_liable.py:15]: The tax form points to the filing requirements on page 2, which compare gross income to what looks to be the standard deduction. `id_income_tax_before_non_refundable_credits > 0` reasonably approximates that, but let's make a new variable with that formula to clarify that piece of the requirement. Please also file an issue to model the filing requirement accurately.
```suggestion
            tax_unit("id_income_tax_liable", period)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**